resources/lint/profiles/mercury.raml

```diff
@@ -198,3 +199,12 @@ validations:
       core.version:
         pattern: ^v[0-9]+$
         minCount: 1
+
+  #datatype rules
+  upper-camelcase-datatype:
+    message: Data type definitions should be UpperCamelCase
+    targetClass: doc.Unit
+    propertyConstraints:
+      doc.declares/shacl.name:
+        pattern: ^[A-Z]+([A-Z]?[a-z0-9]?)*$
```

Comment:
```suggestion
        pattern: ^[A-Z][A-Za-z0-9]*$
```
This is the simplified version of the same pattern

Reply:
Done